frontend/src/metabase/hoc/Toast.jsx

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import { connect } from "react-redux";
+
+import { createUndo, addUndo } from "metabase/redux/undo";
+
+const mapDispatchToProps = {
+  createUndo,
+  addUndo,
+};
+
+const withToast = ComposedComponent => {
+  @connect(null, mapDispatchToProps)
+  class ToastedComponent extends React.Component {
+    _triggerToast(toastContent) {
```

Comment:
I've been preferring `_triggerToast = (toastContent) => {` then you can get rid of the `.bind(this)` in render, that way it's just bound once on instantiation.